sdk/core/Azure.Core/src/Shared/NextLinkOperationImplementation.cs

```diff
@@ -356,6 +356,15 @@ private static bool IsFinalState(Response response, HeaderSource headerSource, o
             return true;
         }
 
+        private static string GetRequiredString(in JsonElement element)
+        {
+            var value = element.GetString();
+            if (value == null)
+                throw new InvalidOperationException($"The requested operation requires an element of type 'String', but the target element has type '{element.ValueKind}'.");
```

Comment:
Should this be `JsonException`?

Reply:
He is just copying what was there before to maintain back compat on behavior.

https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/src/Shared/JsonElementExtensions.cs#L109